src/GitVersionCore/LibGit2/GitObjects.cs

```diff
@@ -0,0 +1,364 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using GitVersion.Helpers;
+using LibGit2Sharp;
+
+namespace GitVersion
+{
+    public class ObjectId
```

Comment:
> interface and that the implementation was moved to its own assembly so we could remove the dependency on LibGit2Sharp from `GitVersionCore` altogehter.

That's exactly what I was thinking of. As next step of this refactoring will be to move the implementation out of `GitVersionCore`, so that we can later have a second managed implementation. I will do create an issue that points to this comment, as context